include/webserver.h

```diff
@@ -69,200 +84,73 @@ class CWebServer
     };
 
     AsyncWebServer _server;
+    StaticStatistics _staticStats;
 
-  public:
-
-    CWebServer()
-        : _server(80)
-    {
-    }
+    // Helper functions/templates
+    template<typename Tv>
+    using ParamValueGetter = std::function<Tv(AsyncWebParameter *param)>;
 
-    // AddCORSHeaderAndSendOKResponse
-    //
-    // Sends an empty OK/200 response; normally used to finish up things that don't return anything, like "NextEffect"
+    template<typename Tv>
+    using ValueSetter = std::function<void(Tv)>;
 
-    void AddCORSHeaderAndSendOKResponse(AsyncWebServerRequest * pRequest, const char * strText = nullptr)
+    template<typename Tv>
+    static void PushPostParamIfPresent(AsyncWebServerRequest * pRequest, const String &paramName, ValueSetter<Tv> setter, ParamValueGetter<Tv> getter)
     {
-        AsyncWebServerResponse * pResponse = (strText == nullptr) ? 
-                                                    pRequest->beginResponse(200) :
-                                                    pRequest->beginResponse(200, "text/json",  strText);    
-        pResponse->addHeader("Access-Control-Allow-Origin", "*");
-        pRequest->send(pResponse);      
-    }
+        if (!pRequest->hasParam(paramName, true, false))
+            return;
 
-    void GetEffectListText(AsyncWebServerRequest * pRequest)
-    {
-        static size_t jsonBufferSize = JSON_BUFFER_BASE_SIZE;
-        bool bufferOverflow;
-        debugV("GetEffectListText");
+        debugV("found %s", paramName.c_str());
 
-        do 
-        {
-            bufferOverflow = false;
-            std::unique_ptr<AsyncJsonResponse> response(new AsyncJsonResponse(false, jsonBufferSize));
-            response->addHeader("Server","NightDriverStrip");
-            auto j = response->getRoot();
-
-            j["currentEffect"]         = g_aptrEffectManager->GetCurrentEffectIndex();
-            j["millisecondsRemaining"] = g_aptrEffectManager->GetTimeRemainingForCurrentEffect();
-            j["effectInterval"]        = g_aptrEffectManager->GetInterval();
-            j["enabledCount"]          = g_aptrEffectManager->EnabledCount();
-
-            for (int i = 0; i < g_aptrEffectManager->EffectCount(); i++) 
-            {
-                DynamicJsonDocument effectDoc(256);
-                effectDoc["name"]    = g_aptrEffectManager->EffectsList()[i]->FriendlyName();

```

Comment:
Not a big deal, but I think EffectsList returns the entire array of effects each time we index into it.  Might be worth having it take a parameter of which effect and fetching just the one?

Reply:
It depends. The endpoint is currently used by the current web interface, which retrieves and uses the data of the whole list. We could fetch and retain the effects list once before the loop, to reduce the number of EffectsList() invocations.